src/ui_tools/interaction.rs

```diff
@@ -17,14 +19,21 @@ pub struct InteractionPalette {
     pub pressed: Color,
 }
 
-fn apply_interaction_palette(
+fn apply_interaction(
     mut palette_query: InteractionQuery<(&InteractionPalette, &mut BackgroundColor)>,
+    mut commands: Commands,
 ) {
     for (interaction, (palette, mut background)) in &mut palette_query {
         *background = match interaction {
             Interaction::None => palette.none,
-            Interaction::Hovered => palette.hovered,
-            Interaction::Pressed => palette.pressed,
+            Interaction::Hovered => {
+                commands.trigger(Sfx::ButtonHover);
+                palette.hovered
```

Comment:
These should not be coupled. Trigger the SFX in a separate system without `InteractionPalette` / `BackgroundColor` in the query, and filter for `With<Button>` or a custom marker component.

System names can be `apply_interaction_palette` and `trigger_interaction_sfx`, for example.